### PR TITLE
Add optional support for ANSI formatting

### PR DESCRIPTION
**Pull request by maximih**
I have enhanced my old PR to use an optional feature to enable ANSI formatting support.During implementation, I noticed `tabwriter.rs` is no longer needed so I removed it.

### CODE REVIEW

**Comment by maximih** [comment]
Ping @BurntSushi, in case you mist my new/revamped PR.

I made the ANSI formatting support an optional feature.The benefit with an optional feature is that the ansi_formatting related dependencies are not pulled if the feature is not enabled.

 Do you like this approach?

File: src/lib.rs [hunk 1]
```diff
@@ -301,3 +303,36 @@ fn display_columns(bytes: &[u8]) -> usize {
...
+        }
+    }
+}
+
+#[cfg(feature = "ansi_formatting")]
+fn strip_formatting(input: &str) -> Option<String> {
+    // check if the input actually contains ANSI escape codes
+    // to avoid unnecessary allocations
+    if input.contains("\x1b[") {
+        let re = regex::Regex::new("\x1B\\[.+?m").unwrap();
```
Comment (BurntSushi) [inline src/lib.rs]: Hmm. So this is going to re-compile the regex on every single call to `strip_formatting`. The easiest way to work around this is to probably use the `lazy_static!` crate. I think something like this should work:

```rust
fn strip_formatting(input: &str) -> Option<String> {
    lazy_static! {
        static ref re: regex::Regex = Regex::new("\x1B\\[.+?m").unwrap();
    }
}
```

**Comment by BurntSushi** [comment]
@maximih Thanks for the reminder! I have one small nit and then I'd be happy to take this! Thank you!

**Comment by maximih** [comment]
@BurntSushi Updated with your suggestion. 

Regarding the tests, it would be nice to run travis also with `ansi_formatting` feature enabled. I guess that is something you need to change.

**Comment by BurntSushi** [comment]
Awesome, thank you. Sorry I've been such a pain about getting this merged, but thank you for your patience!

**Comment by BurntSushi** [comment]
Good point about Travis. I'll take care of it now and push out a new release to crates.io. :-)

**Comment by BurntSushi** [comment]
Done!

**Pull request merged**